modules/siddhi-query-api/src/main/java/org/wso2/siddhi/query/api/execution/query/input/state/State.java

```diff
@@ -51,6 +51,12 @@ public static StateElement logicalOr(StreamStateElement streamStateElement1,
         return new LogicalStateElement(streamStateElement1, LogicalStateElement.Type.OR, streamStateElement2, time);
     }
 
+    public static StateElement not(BasicSingleInputStream basicSingleInputStream,
```

Comment:
Just using `logicalNot` is okay. But using `LogicalStateElement` instead of `AbsentStreamStateElement` will introduce unnecessary code duplicate and complex logic in `StateInputStreamParser` because it is much closer to the `StateElement` than `LogicalStateElement`.
If it is still worth to use the existing structure, I am okay to change it. WDYT?
Thanks